core/templates/domain/statistics/ExplorationStatsObjectFactory.ts

```diff
@@ -67,6 +69,37 @@ export class ExplorationStats {
     }
     return this.stateStatsMapping.get(stateName);
   }
+
+  withStateAdded(newStateName: string): ExplorationStats {
+    const newStateStatsMapping = new Map(this.stateStatsMapping);
+    newStateStatsMapping.set(newStateName, new StateStats(0, 0, 0, 0, 0, 0));
+    return new ExplorationStats(
+      this.expId, this.expVersion, this.numStarts, this.numActualStarts,
+      this.numCompletions, newStateStatsMapping);
+  }
+
+  withStateDeleted(oldStateName: string): ExplorationStats {
+    const newStateStatsMapping = new Map(this.stateStatsMapping);
+    // ES2016 Map uses delete as a method name despite it being a reserved word.
```

Comment:
Did you file a bug with them? Please link to it from here.

Reply:
Right, the pre commit linter raises an error.